### ISSUE TITLE
Add git commit hash etc to log

### ISSUE THREAD

**Issue by ewallace** [riboviz/RiboViz#90]
Currently (commit `3b385d5`), the log file `riboviz.date-time.log` notes the *python* version it was run under. It might be additionally helpful to know which version of riboviz ran it, i.e. to log the commit hash in the log file. That would make it easier for users who've run riboviz on a dataset to return and run an identical version, i.e. reproduce results.

Should we also log versions of any other dependencies?

**Comment by mikej888** [comment]
I've a script in install/environment.sh which dumps info on versions of OS, Python, R (and their packages) and third-party tools. This could be migrated to a Python function.

**Comment by mikej888** [comment]
Relates to #123

**Comment by mikej888** [comment]
Branch [provenance-90-115-123](https://github.com/riboviz/RiboViz/tree/provenance-90-115-123) branched from [remove-sample-prefix-114](https://github.com/riboviz/RiboViz/tree/remove-sample-prefix-114) addresses this plus elated tickets #115, #123.

**Comment by mikej888** [comment]
Commits to 6dd1cd4 updates Python and R tools to invoke new `provenance.py`/`provenance.R` `get_version` functions which get commit hash and date. These use [gitpython](https://gitpython.readthedocs.io/en/stable/) and [git2r](https://docs.ropensci.org/git2r/) respectively. For example:

```console
$ PYTHONPATH=~/riboviz python -m riboviz.tools.demultiplex_fastq
demultiplex_fastq.py version commit 8e6c97e4fbae2892e66616e2fcec0d60416d1f22 date 2019-12-20 02:56:02-08:00
...
$ Rscript riboviz/rscripts/bam_to_h5.R -h
[1] "bam_to_h5.R version commit 611240029d02b62b697a3d05c948611c4a0a836d date 2019-12-20 12:46:25"
...
```

Example if run in RiboViz directory with no `.git` repository folder:

```console
$ PYTHONPATH=~/riboviz-nogit/ python -m riboviz.tools.demultiplex_fastq
demultiplex_fastq.py version unknown
...

$ Rscript riboviz-nogit/rscripts/bam_to_h5.R -h
[1] "bam_to_h5.R version unknown"
...
```

**Comment by mikej888** [comment]
#90, #115, #123 are done - will create pull request in new year.

**Comment by mikej888** [comment]
Merged via pull request #126.
